# HG changeset patch
# User Olaf Till <i7tiol@t-online.de>
# Date 1388759673 -3600
#      Fri Jan 03 15:34:33 2014 +0100
# Branch stable
# Node ID 442bca7dea44e699730a9606d1d3303bcef9554e
# Parent  8d7e45ea11fa416ee2f9ca6deb5bcb5834c22a47
Fix installing packages where dependency name contains '-' (bug #41087).

* scripts/pkg/private/fix_depends.m: Add '-' to list of allowed
characters in a name.

diff --git a/scripts/pkg/private/fix_depends.m b/scripts/pkg/private/fix_depends.m
--- a/scripts/pkg/private/fix_depends.m
+++ b/scripts/pkg/private/fix_depends.m
@@ -24,17 +24,17 @@
 
 ## Make sure the depends field is of the right format.
 ## This function returns a cell of structures with the following fields:
 ##   package, version, operator
 function deps_cell = fix_depends (depends)
   deps = strtrim (ostrsplit (tolower (depends), ","));
   deps_cell = cell (1, length (deps));
   dep_pat = ...
-  '\s*(?<name>\w+)+\s*(\(\s*(?<op>[<>=]+)\s*(?<ver>\d+\.\d+(\.\d+)*)\s*\))*\s*';
+  '\s*(?<name>[-\w]+)\s*(\(\s*(?<op>[<>=]+)\s*(?<ver>\d+\.\d+(\.\d+)*)\s*\))*\s*';
 
   ## For each dependency.
   for i = 1:length (deps)
     dep = deps{i};
     [start, nm] = regexp (dep, dep_pat, 'start', 'names');
     ## Is the dependency specified 
     ## in the correct format?
     if (! isempty (start))
